x-pack/plugins/ml/server/models/model_management/models_provider.ts

```diff
@@ -446,18 +448,39 @@ export class ModelsProvider {
       }
     }
 
-    const result = Object.entries(ELASTIC_MODEL_DEFINITIONS).map(([name, def]) => {
+    const modelDefinitionMap = new Map<string, ModelDefinitionResponse[]>();
+
+    for (const [name, def] of Object.entries(ELASTIC_MODEL_DEFINITIONS)) {
       const recommended =
         (isCloud && def.os === 'Linux' && def.arch === 'amd64') ||
         (sameArch && !!def?.os && def?.os === osName && def?.arch === arch);
-      return {
-        ...def,
-        name,
+
+      const { modelName, ...rest } = def;
+
+      const modelDefinitionResponse = {
+        ...rest,
         ...(recommended ? { recommended } : {}),
+        name,
       };
-    });
 
-    return result;
+      if (modelDefinitionMap.has(modelName)) {
+        modelDefinitionMap.get(modelName)!.push(modelDefinitionResponse);
+      } else {
+        modelDefinitionMap.set(modelName, [modelDefinitionResponse]);
+      }
+    }
+
+    // check if there is no recommended, so we mark default as recommended
+    for (const [, arr] of modelDefinitionMap.entries()) {
```

Comment:
Nit
```suggestion
    for (const arr of modelDefinitionMap.values()) {
```

Reply:
Refactored in https://github.com/elastic/kibana/pull/167407/commits/beef167d516e35179858be75a40ca33d0b221775